File: ScaleformUI_Csharp/Menus/UIMenu/Panels/UIMenuPercentagePanel.cs
using CitizenFX.Core;
using CitizenFX.Core.Native;
using System.Drawing;

namespace ScaleformUI.Menu
{
    public class UIMenuPercentagePanel : UIMenuPanel
    {

        public string Min { get; set; }
        public string Max { get; set; }
        public string Title { get; set; }

        public event PercentagePanelChangedEvent OnPercentagePanelChange;

        internal float _value;

        public float Percentage
        {
            get
            {
                return _value;
            }
            set
            {
                _value = value;
                if (ParentItem != null && ParentItem.Parent != null && ParentItem.Parent.Visible)
                {
                    int it = ParentItem.Parent.MenuItems.IndexOf(ParentItem);
                    ParentItem.Parent.SendPanelsToItemScaleform(it, true);
                    PercentagePanelChange();
                    ParentItem.Parent.PercentagePanelChange(ParentItem, this, _value);
                }
            }
        }


        public UIMenuPercentagePanel(string title, string MinText = "0%", string MaxText = "100%", float initialValue = 0)
        {
            Min = MinText;
            Max = MaxText;
            Title = !string.IsNullOrWhiteSpace(title) ? title : "Opacity";
            _value = initialValue;
        }

        internal void PercentagePanelChange()
        {
            OnPercentagePanelChange?.Invoke(ParentItem, this, Percentage);
        }
    }
}

File: ScaleformUI_Lua/src/Menus/UIMenu/panels/UIMenuPercentagePanel.lua
UIMenuPercentagePanel = setmetatable({}, UIMenuPercentagePanel)
UIMenuPercentagePanel.__index = UIMenuPercentagePanel
UIMenuPercentagePanel.__call = function() return "UIMenuPanel", "UIMenuPercentagePanel" end

---@class UIMenuPercentagePanel
---@field public Min string
---@field public Max string
---@field public Title string
---@field public Percentage number
---@field public ParentItem table
---@field public SetParentItem fun(self:UIMenuStatisticsPanel, item:UIMenuItem):UIMenuItem -- required
---@field public OnPercentagePanelChange function

---New
---@param title string
---@param minText string
---@param maxText string
---@param initialValue number
function UIMenuPercentagePanel.New(title, minText, maxText, initialValue)
    local _UIMenuPercentagePanel = {
        Min = minText or "0%",
        Max = maxText or "100%",
        Title = title or "Opacity",
        _percentage = initialValue or 0.0,
        ParentItem = nil, -- required
        OnPercentagePanelChange = function(item, panel, value)
        end
    }
    return setmetatable(_UIMenuPercentagePanel, UIMenuPercentagePanel)
end

function UIMenuPercentagePanel:Percentage(value)
    if value ~= nil then
        self._percentage = value
        if self.ParentItem ~= nil and self.ParentItem:SetParentMenu() ~= nil and self.ParentItem:SetParentMenu():Visible() then
            local it = IndexOf(self.ParentItem:SetParentMenu().Items, self.ParentItem)
            self.ParentItem:SetParentMenu():SendPanelsToItemScaleform(it, true)
            self.OnPercentagePanelChanged(self.ParentItem, self, self._percentage)
            self.ParentItem:SetParentMenu().OnPercentagePanelChange(self.ParentItem, self, self._percentage)
        end
    else
        return self._percentage
    end
end

---SetParentItem
---@param Item table
function UIMenuPercentagePanel:SetParentItem(Item) -- required
    if Item ~= nil then
        self.ParentItem = Item
    else
        return self.ParentItem
    end
end
